### PR TITLE
feat(telemetry): Add first version of the telemetry module

### PR DESCRIPTION
**Pull request by hadrienk**
This experiment wraps the datadog statsd as a MetricExporter. This allows us to use the API of the MetricProvider right away.

### CODE REVIEW

File: .github/workflows/golang-ci.yaml [hunk 1]
```diff
@@ -9,7 +9,7 @@ jobs:
   matrix-prep:
     runs-on: ubuntu-latest
     outputs:
-      paths: '["pkg"]'
+      paths: '["pkg", "telemetry"]'
```
Comment (alfredosa) [inline .github/workflows/golang-ci.yaml:12]: nice 😋

File: telemetry/dd/metric.go [hunk 2]
```diff
@@ -0,0 +1,254 @@
...
+
+	"github.com/DataDog/datadog-go/v5/statsd"
+	"github.com/iancoleman/strcase"
+	"go.opentelemetry.io/otel/attribute"
+	sdkmetric "go.opentelemetry.io/otel/sdk/metric"
+	"go.opentelemetry.io/otel/sdk/metric/metricdata"
+	semconv "go.opentelemetry.io/otel/semconv/v1.17.0"
+)
+
+var _ sdkmetric.Exporter = *(*ddMetricExporter)(nil)
```
Comment (alfredosa) [inline telemetry/dd/metric.go:18]: is this a hack?

File: telemetry/doc.go [hunk 3]
```diff
@@ -0,0 +1,2 @@
+// Package telemetry contains helpers and tools related to telemetry.
+package telemetry
```
Comment (alfredosa) [inline telemetry/doc.go:2]: placeholder?

File: telemetry/dd/metric.go [hunk 4]
```diff
@@ -0,0 +1,254 @@
...
+		if i+1 < len(p.Bounds) {
+			lower, upper = p.Bounds[i], p.Bounds[i+1]
+		} else {
+			lower, upper = p.Bounds[i], math.Inf(1)
+		}
+		bounds := []attribute.KeyValue{
+			attribute.String("lower_bound", fmt.Sprintf("%f", lower)),
+			attribute.String("upper_bound", fmt.Sprintf("%f", upper)),
+		}
+		err = errors.Join(err, d.client.Count(fmt.Sprintf("%s.bucket", name), int64(p.BucketCounts[i]), toTags(tags, p.Attributes.ToSlice(), bounds), 1.0))
```
Comment (alfredosa) [inline telemetry/dd/metric.go:243]: hehehe

File: telemetry/doc.go (hunk 3 above)
Comment (hadrienk) [reply telemetry/doc.go:2]: Yeah, I've got more cookin'

**Review by alfredosa — APPROVED** [review]

**Pull request merged**